lock-api/src/main/java/com/palantir/lock/client/BlockEnforcingLockService.java

```diff
@@ -46,6 +46,13 @@
  * Fairness is admittedly compromised, but this is a closer approximation than the previous behaviour.
  */
 final class BlockEnforcingLockService {
+    /**
+     * Bound lock acquire timeout by client read timeout because 1. the async request expires before server cancels
+     * the request thus avoiding {@link java.util.concurrent.CancellationException} and
+     * 2. this ensures locks are not given out to phantom client.
+     */
+    private static final Duration MAX_PERMISSIBLE_LOCK_ACQUIRE_TIMEOUT = Duration.ofSeconds(65);
+
```

Comment:
From Jeremy - exactly 65 won’t quite work, because a client’s connect timeout will also cover the time timelock takes to deserialise the request, check it is the leader